spec/controllers/events_controller_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.describe Api::EventsController, type: :controller do
+  let(:event_attributes) { %w[id name date description status user_id] }
+
+  describe 'GET #index' do
+    it 'should get index' do
+      get :index
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'GET #show' do
+    context 'when valid' do
+      let(:event) { create(:event) }
+      it 'returns json response with event' do
+        get :show, params: { id: event.id }
+        json_response = JSON.parse(response.body)
+
+        expect(json_response.keys).to eq(event_attributes)
+        expect(json_response['name']).to eq(event.name)
+      end
+    end
+
+    context 'when invalid' do
+      it 'return unprocessable_entity status' do
+        get :show, params: { id: -1 }
+        json_response = JSON.parse(response.body)
+
+        expect(json_response['status']).to eq('unprocessable_entity')
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when valid' do
+      let(:valid_params) do
+        { name: 'first event',
+          description: 'it is enough long description',
+          status: 'social',
+          date: '2019-11-08T05:00' }
+      end
+      it 'return json with params of created event' do
+        post :create, params: { event: valid_params }
+        json_response = JSON.parse(response.body)
+        event = assigns(:event)
+        expect(event.name).to eq(valid_params[:name])
+        expect(json_response.keys).to eq(event_attributes)
+        expect(json_response['name']).to eq(valid_params[:name])
+      end
+    end
+
+    context 'when invalid' do
+      let(:invalid_params) do
+        { name: nil,
+          description: 'so short',
+          status: 'social',
+          date: '2019-11-08T05:00' }
+      end
+      it 'return message errors' do
+        post :create, params: { event: invalid_params }
+        event = assigns(:event)
+        expect(event.errors.messages[:name].first).to eq('can\'t be blank')
```

Comment:
maybe better to check errors which we get from response, not from event object from controller?